vmdb/app/models/ems_refresh/save_inventory_cloud.rb

```diff
@@ -79,95 +79,41 @@ def save_ems_cloud_inventory(ems, hashes, target = nil)
 
   def save_flavors_inventory(ems, hashes, target = nil)
     return if hashes.nil?
-    target = ems if target.nil?
-
-    ems.flavors(true)
-    deletes = if (target == ems)
-      ems.flavors.dup
-    else
-      []
-    end
-
-    self.save_inventory_multi(:flavors, Flavor, ems, hashes, deletes, :ems_ref)
+    save_inventory_multi(:flavors, Flavor, ems, hashes, (target.nil? || target == ems), :ems_ref)
```

Comment:
Or perhaps, since this is super common across all the methods, extract to a method with a good method name.